lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -23,11 +23,12 @@ class TestResultsPush:
     URL = "http://pbench.example.com/api/v1"
 
     @staticmethod
-    def add_http_mock_response(code: HTTPStatus = HTTPStatus.OK):
+    def add_http_mock_response(code: HTTPStatus = HTTPStatus.OK, message: str = ""):
         responses.add(
             responses.PUT,
             f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
             status=code,
+            json={"message": message},
```

Comment:
Given the `msg = res.text if res.text else res.reason`, I think we want to be able to mock a real `json["message"]`, an empty `json["message"]` (which are both handled here) and also a missing JSON response and maybe a text response that's not `json=` to try all the cases.

As Webb mentioned, a parametrized test would allow going through the list easily.